spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/message-consumption.adoc

```diff
@@ -0,0 +1,1030 @@
+
+`DefaultPulsarMessageListenerContainer` creates only a single consumer.
+If you want to have multiple consumers managed through multiple threads, you need to use `ConcurrentPulsarMessageListenerContainer`.
+
+=== ConcurrentPulsarMessageListenerContainer
+
+`ConcurrentPulsarMessageListenerContainer` has the following constructor:
+
+[source, java]
+----
+public ConcurrentPulsarMessageListenerContainer(PulsarConsumerFactory<? super T> pulsarConsumerFactory,
+    PulsarContainerProperties pulsarContainerProperties)
+----
+
+`ConcurrentPulsarMessageListenerContainer` lets you specify a `concurrency` property through a setter.
+Concurrency of more than `1` is allowed only on non-exclusive subscriptions (`failover`, `shared`, and `key-shared`).
+You can only have the default `1` for concurrency when you have an exclusive subscription mode.
+
+The following example enables `concurrency` through the `PulsarListener` annotation for a `failover` subscription.
+
+[source, java]
+----
+@PulsarListener(topics = "my-topic", subscriptionName = "subscription-1",
+				subscriptionType = SubscriptionType.Failover, concurrency = "3")
+void listen(String message, Consumer<String> consumer) {
+    ...
+    System.out.println("Current Thread: " + Thread.currentThread().getName());
+    System.out.println("Current Consumer: " + consumer.getConsumerName());
+}
+----
+
+In the preceding listener, it is assumed that the topic `my-topic` has three partitions.
+If it is a non-partitioned topic, having concurrency set to `3` does nothing. You get two idle consumers in addition to the main active one.
+If the topic has more than three partitions, messages are load-balanced across the consumers that the container creates.
+If you run this `PulsarListener`, you see that messages from different partitions are consumed through different consumers, as implied by the thread name and consumer names printouts in the preceding example.
+
+NOTE: When you use the `Failover` subscription this way on partitioned topics, Pulsar guarantees message ordering.
+
+The following listing shows another example of `PulsarListener`, but with `Shared` subscription and `concurrency` enabled.
+
+[source, java]
+----
+@PulsarListener(topics = "my-topic", subscriptionName = "subscription-1",
+				subscriptionType = SubscriptionType.Shared, concurrency = "5")
+void listen(String message) {
+    ...
+}
+----
+
+In the preceding example, the `PulsarListener` creates five different consumers (this time, we assume that the topic has five partitions).
+
+NOTE: In this version, there is no message ordering, as `Shared` subscriptions do not guarantee any message ordering in Pulsar.
+
+If you need message ordering and still want a shared subscription types, you need to use the `Key_Shared` subscription type.
+
+=== Message Consumption
```

Comment:
The default id of this section is "message-consumption" and because the title has the same section id, the links will end up being scoped as "message-consumption-1" and "message-consumption-2". 
To avoid any confusion, I would suggest we either put a section id on this or rename it to disambiguate it. I am in favor of the former option and give it a name of "Consuming records". Wdyt?

Reply:
Nice catch!
And I liked the idea of renaming it to Consuming records with the similar id. Will make the change

Reply:
Made the changes, please have a look